src/lib/services-manager/widgets/services_table.rb

```diff
@@ -0,0 +1,311 @@
+      # @return [Yast2::SystemService, nil] nil if the service is not found
+      def selected_service
+        ServicesManagerService.find(selected_service_name)
+      end
+
+    private
+
+      DEFAULT_ID = :services_table
+      private_constant :DEFAULT_ID
+
+      # @return [Array<String>] services shown in the table
+      attr_reader :services_names
+
+      # Table widget id
+      #
+      # @return [Yast::Term]
+      def id
+        Id(@id)
+      end
+
+      # Table header
+      #
+      # @return [Yast::Term]
+      def header
+        Header(
+          *columns.map { |c| send("#{c}_title") }
+        )
+      end
+
+      # Content of the table
+      #
+      # @return [Array<Yast::Term>]
+      def items
+        services_names.map { |s| Item(*values_for(s)) }
+      end
+
+      # Values to show in the table for a specific service
+      #
+      # @param service_name [String]
+      # @return [Array<Yast::Term, String>]
+      def values_for(service_name)
+        [row_id(service_name)] + columns.map { |c| send("#{c}_value", service_name) }
+      end
+
+      # Columns to show in the table
+      #
+      # @return [Array<Symbol>]
+      def columns
+        [:name, :start_mode, :state, :description]
+      end
+
+      # Title for name column
+      #
+      # @return [String]
+      def name_title
+        _('Service')
+      end
+
+      # Title for start_mode column
+      #
+      # @return [String]
+      def start_mode_title
+        _('Start')
+      end
+
+      # Title for state column
+      #
+      # @return [String]
+      def state_title
+        _('State')
+      end
+
+      # Title for description column
+      #
+      # @return [String]
+      def description_title
+        _('Description')
+      end
+
+      # Id for a table row of a service
+      #
+      # @param service_name [String]
+      # @return [Yast::Term]
+      def row_id(service_name)
+        Id(service_name)
+      end
+
+      # Value for the name column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def name_value(service_name)
+        max_width = max_column_width(:name)
+        return service_name if service_name.size < max_width
+
+        service_name[0..(max_width - 3)] + "..."
+      end
+
+      # Value for the start_mode column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def start_mode_value(service_name)
+        ServicesManagerService.start_mode_to_human_for(service_name)
+      end
+
+      # Value for the state column of a service
+      #
+      # @param service_name [String]
+      # @return [String]
+      def state_value(service_name)
+        state = TRANSLATIONS[:service_state][service_state(service_name)]
+        substate = TRANSLATIONS[:service_substate][service_substate(service_name)]
+
+        return _(state) unless substate
+
+        format(_("%{state} (%{substate})"), state: _(state), substate: _(substate))
```

Comment:
please add comment for translators. This is quite confusing string marked for translate, so we should try to explain it ideally with example.